epaint/src/tessellator.rs

```diff
@@ -179,33 +179,23 @@ impl Path {
 
 pub mod path {
     //! Helpers for constructing paths
+    use crate::shape::Rounding;
+
     use super::*;
 
     /// overwrites existing points
-    pub fn rounded_rectangle(path: &mut Vec<Pos2>, rect: Rect, corner_radius: f32) {
+    pub fn rounded_rectangle(path: &mut Vec<Pos2>, rect: Rect, corner_radius: Rounding) {
         path.clear();
 
         let min = rect.min;
         let max = rect.max;
 
-        let cr = corner_radius
-            .min(rect.width() * 0.5)
-            .min(rect.height() * 0.5);
-
-        if cr <= 0.0 {
```

Comment:
This is an important common-case optimization. Please add it back! (`if cr == Rounding::none() { … }`)